### PR TITLE
UI for navigating between folders in current path

### PR DESCRIPTION
**Pull request by hosy**
## Description
Added an UI for fast and easy navigating in current path structure to parent folders.

## Related Issue
#354 

## Motivation and Context
Select a parent folder with one action, without navigating to all ViewControllers.

## How Has This Been Tested?
1. Open a deep nested folder structure
2. Tap on the navigation bar title (current folder name)
3. Select a folder in the popup view
4. Content of selected folder should be visible

## Screenshots (if appropriate):
IMG_8088-2

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/owncloud/ios-app/blob/master/CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: ownCloud/Client/BreadCrumbTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import UIKit
+
+class BreadCrumbTableViewController: StaticTableViewController {
+
+	// MARK: - Constants
+	private let contentWidth = 500
```
Comment (mneuwert) [inline ownCloud/Client/BreadCrumbTableViewController.swift]: @hosy does this `contentWidth` work for all content sizes?

File: ownCloud/Client/BreadCrumbTableViewController.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		}
+
+		var rows : [StaticTableViewRow] = []
+		let counter = pathComp.count
+		var iterator = 2
+		let height = rowHeight * counter - 1
+		self.preferredContentSize = CGSize(width: contentWidth, height: height)
+
+		for (_, currentPath) in pathComp.enumerated().reversed() {
+			let counter = stackViewControllers.count
```
Comment (mneuwert) [inline ownCloud/Client/BreadCrumbTableViewController.swift]: @hosy `counter` can be put outside of the loop

File: ownCloud/Client/BreadCrumbTableViewController.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+		if queryPath.hasSuffix("/") {
+			pathComp.removeLast()
+		}
+		if pathComp.count > 1 {
+			pathComp.removeLast()
+		}
+
+		var rows : [StaticTableViewRow] = []
+		let counter = pathComp.count
```
Comment (mneuwert) [inline ownCloud/Client/BreadCrumbTableViewController.swift]: @hosy I would prefer more descriptive names than just `counter` or `iterator` (so `rowHeight` as an example is much clearer than just `height`), also adding 'er' suffix does make them sounding as object instance names capable of performing complex actions.

**Comment by jesmrec** [comment]
Approved!

**Pull request merged**